presto-main/src/main/java/io/prestosql/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -669,24 +668,34 @@ private void startInBackground(ManagedQueryExecution query)
     private void queryFinished(ManagedQueryExecution query)
     {
         synchronized (root) {
-            if (!runningQueries.contains(query) && !queuedQueries.contains(query)) {
+            if (!runningQueries.keySet().contains(query) && !queuedQueries.contains(query)) {
                 // Query has already been cleaned up
                 return;
             }
-            // Only count the CPU time if the query succeeded, or the failure was the fault of the user
-            if (!query.getErrorCode().isPresent() || query.getErrorCode().get().getType() == USER_ERROR) {
-                InternalResourceGroup group = this;
-                while (group != null) {
-                    group.cpuUsageMillis = saturatedAdd(group.cpuUsageMillis, query.getTotalCpuTime().toMillis());
-                    group = group.parent.orElse(null);
-                }
-            }
-            if (runningQueries.contains(query)) {
+
+            if (runningQueries.keySet().contains(query)) {
+                ResourceUsage cachedUsage = runningQueries.get(query);
+                // cpu usage is tracked in a cumulative sense over time, memory usage relates to a particular time instant.
```

Comment:
Nit: To be consistent, capitalize CPU.

Perhaps something like this is a little clearer:
```
// CPU is measured cumulatively (i.e. total query CPU used until this moment)
// Memory is measured instantaneously (how much memory the query is using at this moment)